src/ReinforcementLearningCore/src/policies/agent/agent_base.jl

```diff
@@ -17,7 +17,7 @@ mutable struct Agent{P,T} <: AbstractPolicy
     policy::P
     trajectory::T
 
-    function Agent(policy::P, trajectory::T) where {P,T}
+    function Agent(policy::AbstractPolicy, trajectory::Trajectory)
```

Comment:
Is this more efficient regarding typing? Or is it simply more readable?

Reply:
`function Agent(policy::P, trajectory::T) where {P,T} ` was equivalent to `function Agent(policy::Any, trajectory::Any)`, or to `function Agent(policy, trajectory)`. 

Agent expects an abstract policy and a trajectory so it's more robust to user mistakes. Efficiencywise, I reckon it makes no difference at all.

Reply:
Nevermind, P and T were needed for the new{P,T} constructor below.